test/e2e/deployment.go

```diff
@@ -613,24 +613,14 @@ func testRollbackDeployment(f *Framework) {
 	Expect(err).NotTo(HaveOccurred())
 	defer stopDeployment(c, f.Client, ns, deploymentName)
 
-	// Check that deployment is created fine.
-	deployment, err := c.Extensions().Deployments(ns).Get(deploymentName)
+	// Wait for DeploymentStatus to be appropriately updated
+	deploymentReplicasPtr := util.IntPtr(deploymentReplicas)
+	err = waitForDeploymentStatusUpdated(c, ns, deploymentName, deploymentReplicasPtr, deploymentReplicasPtr, deploymentReplicasPtr, util.IntPtr(0))
 	Expect(err).NotTo(HaveOccurred())
 
-	// Verify that the required pods have come up.
-	err = verifyPods(unversionedClient, ns, "nginx", false, deploymentReplicas)
```

Comment:
I think there is value in letting this check be.
Verifying the status update is good but we should also verify that the pods actually came up.